### ISSUE TITLE
haskellPackage.rdf4h : build failed on missing dependencies

### ISSUE THREAD

**Issue by koslambrou** [NixOS/nixpkgs#86345]
**Describe the bug**
When trying to install anything that depends on haskellPackages.rdf4h, the builder fails, complaining that: Setup: Encountered missing dependencies:algebraic-graphs >=0.3 && <0.5

**To Reproduce**
Try running `nix-env -iA nixos.haskellPackages.rdf4h`

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.0-5-amd64, PureOS, noversion`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(kolam): `"nixpkgs-20.09pre221814.10100a97c89"`
 - nixpkgs: `/home/kolam/.nix-defexpr/channels/nixpkgs`
```

**Comment by koslambrou** [comment]
Tried adding the following in `pkgs/development/haskell-modules/configuration-common.nix`:

```
  rdf4h = dontCheck (super.rdf4h.override { algebraic-graphs = self.algebraic-graphs_0_4; });
```

but algebraic-graphs_0_4 is missing.

I also tried

```
  rdf4h = dontCheck (super.rdf4h.override { 
    algebraic-graphs = pkgs.haskellPackages.callHackage "algebraic-graphs" "0.4" {};
  });
```

but I got the following error:

```
Configuring cabal2nix-2.15.2...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1022:20 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:475:12 in Cabal-3.0.1.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.0.1.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.0.1.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.0.1.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.0.1.0:Distribution.Simple
  defaultMain, called at Setup.hs:6:8 in main:Main
Setup: Encountered missing or private dependencies:
Cabal ==3.2.*
```

I'm stuck here. Any help ?

**Comment by cdepillabout** [comment]
@koslambrou I was able to get this to at least start building on the `haskell-updates` branch at commit f5b9b44a9 with the following change:

```diff
diff --git a/pkgs/development/haskell-modules/configuration-common.nix b/pkgs/development/haskell-modules/configuration-common.nix
index 4181f7dfc19..d377967cac8 100644
--- a/pkgs/development/haskell-modules/configuration-common.nix
+++ b/pkgs/development/haskell-modules/configuration-common.nix
@@ -1500,4 +1500,8 @@ self: super: {
     haskell-lsp = self.haskell-lsp_0_19_0_0;
   });
 
+  rdf4h = dontCheck (super.rdf4h.override {
+    algebraic-graphs = dontCheck (doJailbreak (self.callHackage "algebraic-graphs" "0.4" {}));
+  });
+
 } // import ./configuration-tensorflow.nix {inherit pkgs haskellLib;} self super
```

This couldn't be merged in like this (since you can't use `self.callHackage` in nixpkgs because of IFD), but hopefully it will get you on the right track of figuring this out.

This fails because of MonadFail-related changes, so it doesn't look like rdf4h supports ghc-8.8.  I'd suggest opening a PR upstream that gets rdf4h to support ghc-8.8, and newer versions of `algebraic-graphs`.

**Comment by svmhdvn** [comment]
Should be fixed now, this is closable.

**Comment by cdepillabout** [comment]
@svmhdvn Thanks!

Fixed in #91420.
